src/libraries/Microsoft.PowerFx.Core/Localization/StringResources.cs

```diff
@@ -31,13 +27,16 @@ internal enum ResourceFormat
 
         private const string FallbackLocale = "en-US";
 
+        private static readonly ResourceManager _resourceManager = new ResourceManager("Microsoft.PowerFx.Core.strings.PowerFxResources", typeof(StringResources).Assembly);
```

Comment:
>_resourceManager [](http://example.com/codeflow?start=48&length=16)

This is static, not protected by a lock. 

If ResourceManager is semantically immutable (and thread safe),  then add it to _knownImmutableTypes in AnalyzeThreadSafety.cs. 

Else, we should add a lock.

Reply:
Docs claim it's thread safe
https://docs.microsoft.com/en-us/dotnet/api/system.resources.resourcemanager?view=net-6.0#thread-safety

Reply:
Docs are lying :) 
[https://docs.microsoft.com/en-us/dotnet/api/system.resources.resourcemanager.releaseallresources?view=net-6.0](https://docs.microsoft.com/en-us/dotnet/api/system.resources.resourcemanager.releaseallresources?view=net-6.0)

GetString() is thread safe w.r.t other calls to GetString()
But not safe w.r.t. calls to ReleaseAll()

We don't call ReleaseAll(), so not an issue in practice. 

Not required, but If we really wanted to by "defensive", here, could create a wrapper around ResourceManager that only allows GetString() and declare that wrapper as ThreadSafe.